internal/service/project/resource_project_test.go

```diff
@@ -1,20 +1,441 @@
+
+			if (err != nil) != tc.expectedError {
+				t.Errorf("Case %s: Received unexpected error: %v", tc.name, err)
+			}
+		})
+	}
+}
+
+func TestUpdateProjectTeams(t *testing.T) {
+	teamRoles, _ := types.SetValueFrom(context.Background(), types.StringType, []string{"BASIC_PERMISSION"})
+	teamOne := project.TfTeamModel{
+		TeamID:    types.StringValue("team1"),
+		RoleNames: teamRoles,
+	}
+	teamTwo := project.TfTeamModel{
+		TeamID: types.StringValue("team2"),
+	}
+	teamRolesUpdated, _ := types.SetValueFrom(context.Background(), types.StringType, []string{"ADMIN_PERMISSION"})
+	updatedTeam := project.TfTeamModel{
+		TeamID:    types.StringValue("team1"),
+		RoleNames: teamRolesUpdated,
+	}
+	singleTeamSet, _ := types.SetValueFrom(context.Background(), project.TfTeamObjectType, []project.TfTeamModel{teamOne})
+	twoTeamSet, _ := types.SetValueFrom(context.Background(), project.TfTeamObjectType, []project.TfTeamModel{teamOne, teamTwo})
+	updatedTeamSet, _ := types.SetValueFrom(context.Background(), project.TfTeamObjectType, []project.TfTeamModel{updatedTeam})
+
+	testCases := []struct {
+		name          string
+		mockResponses []ProjectResponse
+		projectState  project.TfProjectRSModel
+		projectPlan   project.TfProjectRSModel
+		expectedError bool
+	}{
+		{
+			name: "Teams has not changed",
+			projectState: project.TfProjectRSModel{
+				Name:  name,
+				Teams: singleTeamSet,
+			},
+			projectPlan: project.TfProjectRSModel{
+				Name:  name,
+				Teams: singleTeamSet,
+			},
+			expectedError: false,
+		},
+		{
+			name: "Add teams",
+			projectState: project.TfProjectRSModel{
+				Name:  name,
+				Teams: singleTeamSet,
+			},
+			projectPlan: project.TfProjectRSModel{
+				Name:  name,
+				Teams: twoTeamSet,
+			},
+			expectedError: false,
+		},
+		{
+			name: "Remove teams",
+			projectState: project.TfProjectRSModel{
+				Name:  name,
+				Teams: twoTeamSet,
+			},
+			projectPlan: project.TfProjectRSModel{
+				Name:  name,
+				Teams: singleTeamSet,
+			},
+			expectedError: false,
+		},
+		{
+			name: "Update teams",
+			projectState: project.TfProjectRSModel{
+				Name:  name,
+				Teams: singleTeamSet,
+			},
+			projectPlan: project.TfProjectRSModel{
+				Name:  name,
+				Teams: updatedTeamSet,
+			},
+			expectedError: false,
+		},
+	}
+
+	for i, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			testObject := new(MockProjectService)
+
+			testObject.On("AddAllTeamsToProject", mock.Anything, mock.Anything, mock.Anything).Return(ProjectResponse{Err: nil})
+			testObject.On("RemoveProjectTeam", mock.Anything, mock.Anything, mock.Anything).Return(ProjectResponse{Err: nil})
+			testObject.On("UpdateTeamRoles", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(ProjectResponse{Err: nil})
+
+			err := project.UpdateProjectTeams(context.Background(), testObject, &testCases[i].projectState, &testCases[i].projectPlan)
```

Comment:
nit: testing this function limits to verifying if there are errors, but not which teams have actually been updated. Looking into the implemenation `getChangesInTeamsSet` could be tested to see which teams are actually being removed, created, and updated.